infra/src/main/kotlin/fi/fta/geoviite/infra/ratko/RatkoService.kt

```diff
@@ -100,13 +102,47 @@ class RatkoService @Autowired constructor(
 
             // Here, we only care about current moment, but fix it to the latest publication DB time, pushed or not
             val latestPublicationMoment = ratkoPushDao.getLatestPublicationMoment()
-            val switchChanges = calculatedChangesService.getAllSwitchChangesByLocationTrackChange(
-                locationTrackChanges = locationTrackChanges,
-                moment = latestPublicationMoment,
-            )
+            val switchChanges = locationTrackChanges
+                .flatMap { locationTrackChange ->
+                    getSwitchChangesByLocationTrack(
+                        locationTrackId = locationTrackChange.locationTrackId,
+                        filterByKmNumbers = locationTrackChange.changedKmNumbers,
+                        moment = latestPublicationMoment
+                    )
+                }.map { switchChange ->
+                    val switch = switchService.getOrThrow(PublishType.OFFICIAL, switchChange.switchId)
```

Comment:
Nämä ei nyt enää kunnioita tuota kiinitetyn ajanhetken versiota vaan hakee uudelleen tuoreimman version. Vaikka tod.näk. on tietenkin pieni, tuossa voi periaatteessa vaihtua versio välistä alta ja mennä sisäisesti ristiriitainen yhdistelmä. Tämä on myös hitaampaa, koska jokainen näin tehty haku käy checkaamassa uudeleen rowversion kannasta omalla roundtripillä.

Eli summa summarum, oikeastaan täälläkin pitäisi oikeastaan pelata vain rowversioilla: kiinnittää ne kerran vaikkapa tuoreimman julkaisun aikaleimalla ja käyttää vain niitä. Tai jos tarvitsee hakea uudelleen, niin hakea rowversio sillä aikaleimalla, ei tuoreinta officialia.